### PR TITLE
feat: support icons in testname format

### LINKED ISSUE #399: why does the testname format does not support icons?
I am reading the codebase after few attempts at finding the incantation to make this work, and took a hint on the fact that in the tests the icons are only tested against the `pkgname` format.

Is there a reason why this doesn't work? Or better, did I miss something in the docs?

### PR DESCRIPTION
**Pull request by kdkasad**
Adds support for `--format-icons` when using `--format testname`.

Notable points:
- The `default` icons are now used if none are specified, rather than `text` like before. This matches the documentation.
- Two spaces are used between the icon and the package name, just like in the `pkgname` format. This applies even when the `text` icons are used, as it seems unwise to add more fragile edge cases.
- Empty packages use the skip icon instead of the pass icon, matching previous behavior.
  - For the `text` icons only, `EMPTY` is used instead of `SKIP` in this case, again matching previous behavior.

Unit tests have been updated/added as appropriate to guarantee the above points are accurate.

The `github-actions` format is unchanged; it required some refactoring to separate from the `testname` format but the behavior stays the same.

Closes #399.

### CODE REVIEW

File: testjson/format.go [hunk 1]
```diff
@@ -76,11 +76,12 @@ func standardJSONFormat(out io.Writer) EventFormatter {
...
-func testNameFormatTestEvent(out io.Writer, event TestEvent) {
+func testNameFormatTestEvent(out io.Writer, event TestEvent, space string, getIcon func(Action) string) {
 	pkgPath := RelativePackagePath(event.Package)
 
-	fmt.Fprintf(out, "%s %s%s (%.2fs)\n",
-		colorEvent(event)(strings.ToUpper(string(event.Action))),
+	fmt.Fprintf(out, "%s%s%s%s (%.2fs)\n",
+		getIcon(event.Action),
+		space,
 		joinPkgToTestName(pkgPath, event.Test),
```
[the remark this answers is not in the document]
Comment (ccoVeille) [reply testjson/format.go:85]: OK then. But I feel strange to see so many golden files updated and see double space everywhere. It seemed to be a problem.

[the remark this answers is not in the document]
Comment (kdkasad) [reply testjson/format.go:85]: Two spaces makes it the same as the `pkgname` format. If you look at the golden files for the `pkgname` format, there are two spaces between the icon (even for `text`) and the package name.

If desired, I can make it so there is only one space between `text` icons and the test name, but that would introduce an edge case that has to be handled separately. It seems easier in terms of maintenance to just use two spaces like the other formats do.

[the remark this answers is not in the document]
Comment (ccoVeille) [reply testjson/format.go:85]: I assumed it was something like that.

I'm fine with current changes now I understood with your replies
